src/action-creators/pull.ts

```diff
@@ -25,22 +28,57 @@ async function itForEach<T>(it: AsyncIterable<T>, callback: (value: T) => void)
   }
 }
 
+/** Returns a list of pending contexts or missing parents from all contextMap contexts and their descendants. */
+const getPendingOrMissingContexts = (state: State, contextMap: Index<Context>) => {
+  /** Returns true if the context is pending or missing from the contextIndex. */
+  const isPendingOrMissing = (context: Context) => {
+    const parent = getParent(state, context)
+    return !parent || parent.pending
+  }
+
+  const contextsFiltered = _.flatMap(Object.values(contextMap), context =>
+    isPendingOrMissing(context)
+      ? // if the original contextMap context is pending, use it
+        [context]
+      : // otherwise search for pending or missing descendants
+        getDescendantContexts(state, context).filter(isPendingOrMissing),
+  )
+
+  return contextsFiltered
+}
+
 /**
  * Fetch, reconciles, and updates descendants of any number of contexts up to a given depth.
  * WARNING: Unknown behavior if thoughtsPending takes longer than throttleFlushPending.
  */
 const pull =
   (
     contextMap: Index<Context>,
-    { maxDepth, onLocalThoughts, onRemoteThoughts }: PullOptions = {},
+    { force, maxDepth, onLocalThoughts, onRemoteThoughts }: PullOptions = {},
   ): Thunk<Promise<boolean>> =>
   async (dispatch, getState) => {
     if (Object.keys(contextMap).length === 0) return false
 
+    let contextMapFiltered = contextMap
+
+    // pull only pending contexts parents unless forced
+    if (!force) {
+      // Missing parents only occur in 2-part deletes (see flushDeletes and associated logic).
```

Comment:
Sure! As you know, in a 2-part delete, all of the descendants that are in the Redux store are deleted in Part I, and pending descendants are pulled and then deleted in Part II. 

With the old style `pull`, Part II pulled the pending descendants as expected. With the new style `pull` that *only* pulls pending thoughts, `pull` will short circuit in Part II since there is no `Parent` which is marked as pending (it was deleted in Part I). Thus we need to check both `pending` and check that the `Parent` is there at all. The only way it could be missing is if it was removed in Part I of a 2-part delete.

Does that make sense?

Reply:
Yes it makes sense.